library/src/main/java/com/datatorrent/lib/formatter/JsonFormatter.java

```diff
@@ -45,58 +38,30 @@
 @org.apache.hadoop.classification.InterfaceStability.Evolving
 public class JsonFormatter extends Formatter<String>
 {
+  private transient ObjectMapper objMapper;
   private transient ObjectWriter writer;
-  protected String dateFormat;
 
+  @SuppressWarnings("deprecation")
   @Override
   public void setup(OperatorContext context)
   {
-    try {
-      ObjectMapper mapper = new ObjectMapper();
-      if (dateFormat != null) {
-        mapper.setDateFormat(new SimpleDateFormat(dateFormat));
-      }
-      writer = mapper.writerWithType(clazz);
-      mapper.configure(SerializationConfig.Feature.AUTO_DETECT_FIELDS, true);
-      mapper.configure(SerializationConfig.Feature.AUTO_DETECT_GETTERS, true);
-      mapper.configure(SerializationConfig.Feature.AUTO_DETECT_IS_GETTERS, true);
-    } catch (Throwable e) {
-      throw new RuntimeException("Unable find provided class");
-    }
+    objMapper = new ObjectMapper();
+    writer = objMapper.writerWithType(clazz);
   }
 
   @Override
   public String convert(Object tuple)
   {
+    if (tuple == null) {
+      return null;
+    }
     try {
       return writer.writeValueAsString(tuple);
-    } catch (JsonGenerationException | JsonMappingException e) {
-      logger.debug("Error while converting tuple {} {}",tuple,e.getMessage());
```

Comment:
logger.debug("Error while converting tuple {} {}",tuple,e); instead of e.getMessage() its in debug mode